### PR TITLE
prefer sqlite blobs to raw

### PR DESCRIPTION
**Pull request by TedSinger**
removing public method PlasticcDatasetLC.get_band - any objections?

### CODE REVIEW

File: justice/lightcurve.py [hunk 1]
```diff
@@ -250,9 +250,30 @@ def get_band(cls, conn, dataset, obj_id, band_id):
...
+        times = np.frombuffer(raw_times, dtype=np.float64)
+        fluxes = np.frombuffer(raw_fluxes, dtype=np.float64)
+        flux_errs = np.frombuffer(raw_flux_errs, dtype=np.float64)
+        detected = np.frombuffer(raw_detected[::8], dtype=np.bool8)
+        return BandData(times, fluxes, flux_errs, detected)
+
     @classmethod
     def _sqlite_get_lc(cls, conn, dataset, obj_id):
-        bands = dict((band, cls.get_band(conn, dataset, obj_id, band_id))
+        conn.execute("select name from sqlite_master where type='table' and name like '{}_blob'".format(dataset))
```
Comment (gatoatigrado) [inline justice/lightcurve.py]: add a comment

File: justice/lightcurve.py [hunk 2]
```diff
@@ -250,9 +250,30 @@ def get_band(cls, conn, dataset, obj_id, band_id):
...
+    @classmethod
+    def _get_band_from_blobs(cls, conn, dataset, obj_id, band_id):
+        res = conn.execute('''
+            select mjd, flux, flux_err, detected
+            from {}_blob
+            where object_id = ?
+            and passband = ?
+            '''.format(dataset), [obj_id, band_id]
+        )
+        raw_times, raw_fluxes, raw_flux_errs, raw_detected = res.fetchone()
```
Comment (gatoatigrado) [inline justice/lightcurve.py]: jw do you have to close res.close()? nbd

Comment (TedSinger) [reply justice/lightcurve.py]: if the connection is passed in by the caller, the caller should decide when to close it. if the caller passes in a string and the function creates the connection, the `with` block handles closing it

Comment (TedSinger) [reply justice/lightcurve.py]: unless i missed something

**Pull request merged**